code/datums/components/larryknife.dm

```diff
@@ -0,0 +1,53 @@
+//Larry With Knives
+/datum/component/larryknife
+    dupe_mode = COMPONENT_DUPE_ALLOWED
+    var/knife_damage
+    var/cooldown
+
+    var/static/list/default_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/knife_crossed,
+	)
+
+/datum/component/larryknife/Initialize(damage = 0)
+    knife_damage = damage
+    RegisterSignal(parent, COMSIG_ATOM_ENTERED, .proc/knife_crossed)
+    RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/knife_move)
+    RegisterSignal(parent, COMSIG_LARRY_DESTROY, .proc/larry_destroyed)
```

Comment:
mob death signal should do the same thing

Reply:
The execution chain of explode() stops well before any mob death signal is called.

Unless QDEL triggers mob death.

Reply:
qdel has its own signal